### ISSUE TITLE
autocomplete userpicker issues

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#6405]
Tested in the theming sandbox (after it failed to work as intended for the Tidypics plugin when trying to tag a user).

On Elgg 1.8.18 both the input-autocomplete and -userpicker fail to show up correctly. When starting to type a username the selection field is either not fully displayed or it gets displayed behind the background (I don't know how to describe it better). You only see one line of grey pixels below the input field which might be the upper border of the selection field.

On Elgg 1.9dev the autocomplete and userpicker work in the theming sandbox. But still it not useable for Tidypics where the autocomplete input is used in a lightbox popup. Here the username selection field (I think this is a lightbox in itself too) is displayed behind the tagging lightbox, so you can't see what names are offered (but only blindly select an entry).

**Comment by iionly** [comment]
Possible solution (or maybe workaround):

Adding "z-index: 10000;" to ui-autocomplete class in views/default/css/elements/forms.php

Feedback?

**Comment by juho-jaakkola** [comment]
I took a look at our current z-index values:
 - lightbox 1100
 - elgg-system-messages 2000
 - elgg-page-topbar 9000
 - elgg-module-popup 9999
 - elgg-menu-hover 10 000

Based to these I suppose 10 000 would be a good value for the autocomplete input.

**Comment by mrclay** [comment]
10000 is fine, but 99999999999999999999 is the only way to be sure. :us: :muscle:

**Comment by jeabakker** [comment]
and to think that 9001 wouldn't even be enough

**Comment by iionly** [comment]
Why 10000?

Originally, the same problem with the autocomplete input also occured on Elgg 1.8 when I opened this issue. For whatever reason it works now on Elgg 1.8 without issue.

On Elgg 1.8 appears a

style="z-index: 10000; ..."

for the ui-autocomplete element. Though I have ABSOLUTELY NO IDEA where it is coming from! On Elgg 1.9 the ui-autocomplete element has no z-index set in its style definition.

Anyway, when adding "z-index: 10000;" to the ui-autocomplete class it works also on Elgg 1.9. I've tested it in the theming sandbox and for the receiver input field on message sending - though for these two cases it also works without the modification. Especially I've tested it on the autocomplete field when tagging an image with Tidypics that itself is within a lightbox - which failed to work correctly without setting the z-index for ui-autocomplete but is fine with the modification.
